### PR TITLE
Doc: Reuse settings content rather than duplicate

### PR DESCRIPTION
**Pull request by karenzone**
Settings content is duplicated in several places in the Logstash Reference, and the content diverges over time. That is, contributors update in content in one place without knowing that the content appears somewhere else.  This work is an attempt to use attribute/value combinations to standardize content and set it up for smart reuse.  

Issue: #14133 

#### More to do:
- [ ]  Changes to  
[Configuration Management Settings in Logstash](https://github.com/elastic/logstash/edit/8.3/docs/static/settings/configuration-management-settings.asciidoc)   (Source file: `/settings/configuration-management-settings.asciidoc`)

**PREVIEW:** https://logstash_14210.docs-preview.app.elstc.co/guide/en/logstash/master/logstash-yml-settings.html

### CODE REVIEW

File: docs/static/settings-file.asciidoc [hunk 1]
```diff
@@ -63,7 +63,17 @@ modules:
...
+[[logstash-yml-settings]]
+=== Logstash settings
+
+The `logstash.yml` file includes these settings. 
+
+[discrete]
+[[settings-pq]]
+==== Persistent queue settings
+
+include::./settings/dlq-settings.asciidoc[]
```
Comment (karenzone) [inline docs/static/settings-file.asciidoc:76]: `ERROR: static/settings-file.asciidoc: line 76: include file not found: /tmp/docsbuild/8gG0huEE8e/logstash/docs/static/settings/dlq-settings.asciidoc`

**Comment by karenzone** [comment]
ToDo:
* Decide how to treat "modules" and the "how to configure modules" section. One option: Move the module config info to the section dedicated to modules.

File: docs/static/settings/pipeline-config-settings.asciidoc [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+
+IMPORTANT: The following `config.field_reference.escape_style` setting is a duplicate of the previous setting while we resolve formatting issues. 
+
+config.field_reference.escape_style:: [Technical preview]. Provides a way to reference fields that contain <<formal-grammar-escape-sequences,field reference special characters>> `[` and `]`. +
+Current options are:   +
+* `percent`: URI-style `%`+`HH` hexadecimal encoding of UTF-8 bytes (`[` -> `%5B`; `]` -> `%5D`). +
+* `ampersand`: HTML-style `&#`+`DD`+`;` encoding of decimal Unicode code-points (`[` -> `&#91;`; `]` -> `&#93;`). +
+* `none`: field names containing special characters _cannot_ be referenced. 
++
+NOTE: This feature is in *technical preview* and may change in the future.
```
Comment (karenzone) [inline docs/static/settings/pipeline-config-settings.asciidoc:27]: @lcawl, here's a section that highlights some of the formatting limitations I'm bumping into with using attributes in a definition list.  Check out `config.field_reference.escape_style` for a good example. 

https://logstash_14210.docs-preview.app.elstc.co/guide/en/logstash/master/logstash-yml-settings.html#settings-pipeline-config

**Comment by karenzone** [comment]
Note to self @karenzone:  Add `dead_letter_queue.retain.age` (logstash.yml and pipelines.yml)

File: docs/include/attributes-ls.asciidoc [hunk 3]
```diff
@@ -8,3 +8,275 @@ with a corresponding change to the VPR settings in
...
+:queuecheckpointretry-desc:	When enabled, Logstash will retry four times per attempted checkpoint write for any checkpoint writes that fail. Any subsequent errors are not retried. +
+This is a workaround for failed checkpoint writes that have been seen only on Windows platform, filesystems with non-standard behavior such as SANs and is not recommended except in those specific circumstances. (queue.type: persisted)+
+Default is `true`.
+
+:queuecheckpointinterval: queue.checkpoint.interval 
+:queuecheckpointinterval-desc: Sets the interval in milliseconds when a checkpoint is forced on the head page (`queue.type: persisted`). Default is 1000. Set to 0 to eliminate periodic checkpoints. 
+
+
+//DLQ settings
+
```
Comment (karenzone) [inline docs/include/attributes-ls.asciidoc:187]: Note to self @karenzone: `Add dead_letter_queue.retain.age` for both logstash.yml and pipelines.yml

File: docs/static/settings/dlq-settings.asciidoc [hunk 4]
```diff
@@ -0,0 +1,11 @@
+// The content for these settings is single sourced from attributes-ls.asciidoc
```
Comment (karenzone) [inline docs/static/settings/dlq-settings.asciidoc:1]: Note to self @karenzone: `Add dead_letter_queue.retain.age` for both logstash.yml and pipelines.yml

File: docs/static/settings/pipeline-config-settings.asciidoc (hunk 2 above)
Comment (lcawl) [reply docs/static/settings/pipeline-config-settings.asciidoc:27]: The [Asciidoctor docs](https://docs.asciidoctor.org/asciidoc/latest/attributes/document-attributes/#what-does-defining-a-document-attribute-mean) seem to say that attributes work if you "have a value that includes basic inline AsciiDoc syntax, such as: ...
text formatting (if wrapped in a [pass macro](https://docs.asciidoctor.org/asciidoc/latest/pass/pass-macro/#inline-pass))...". However, I did some quick tests with `pass:q[formatted values]` and it didn't seem to work. We might not have that functionality enabled in our doc builds. I did, however, get the formatting to work by using tagged regions. If you're interested in that method, see https://github.com/karenzone/logstash/pull/1

**Comment by karenzone** [comment]
Replaced by #14384

File: docs/include/attributes-ls.asciidoc (hunk 3 above)
Comment (karenzone) [reply docs/include/attributes-ls.asciidoc:187]: Included in #14384

**Pull request closed without merging**